### DISCUSSION TITLE
Teensy 4.1 instead of raspberry Pi?

### DISCUSSION THREAD

**Discussion by n3ws** [frntc/RAD#17] [Ideas]
I'm currently using Kung Fu Flash with my C64. It is great, except
that it cannot emulate REU (Sonic).

RAD could do REU, but as I understood cannot emulate other kinds
of cartridges due to a limited number of gpio pins in raspberry pi.

I was wondering would it be possible to do all-in-one expansion
cartridge based on Teensy 4.1? it has 55 IO pins (although some
of those are needed for the SD card etc), possibility to expand ram to 
16 MB, and even an 6510 emulator. Teensy is also generally available, 
unlike the Pi.

16 MB RAM expansion
https://www.pjrc.com/store/psram.html

CPU accelerator
https://www.pjrc.com/commodore-64-accelerator/

**Comment by frntc** [comment]
I have made a CPU+REU emulation (using the MCL64 as hardware-interface, images posted somewhere on Forum64, actually running Sonic) a while ago, but I didn't go further this route. I have never tested if the speed of PSRAM access is sufficient, but if I'd replace the CPU with a Teensy (which requires opening the C64, desoldering), I can do the same (more easily) with the RAD and emulate CPU+cartridges (the Turbo Chameleon does similar things afaik) --- works with and without 6510/8500: https://www.forum64.de/index.php?thread/52749-heute-so-gebastelt/&postID=1937179#post1937179 :-)

I'm using the Teensy for the SIDKick, but I think for projects which requires more resources I better go with a RPi (also in terms of development). The non-availability of RPis is no longer a problem. Since the beginning of the year you can regularly buy RPi 3A+ for decent prices, actually less than a Teensy 4.1 (see e.g. https://rpilocator.com)
